api/app.py

```
@@ -137,30 +186,117 @@ def health():
-        return {"status": "success", "message": "Document ingested."}
-    except HTTPException:
-        raise
-    except Exception as e:
-        logging.exception("Ingest failed")
-        raise HTTPException(status_code=500, detail=str(e))
+    with trace("ingest_doc", run_type="chain",
+               inputs={"id": req.id, "text_len": len(req.text or "")},
+               tags=["ingest", "text"]) as run:
+        try:
+            if not req.text.strip():
+                raise HTTPException(status_code=422, detail="text is required")
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)